sklearn/multioutput.py

```diff
@@ -150,3 +156,134 @@ def score(self, X, y, sample_weight=None):
+                         "score and predict_proba!")
+
+
+def _fit_classification(estimator, X, y):
+    estimator = clone(estimator)
+    estimator.fit(X, y)
+    return estimator
+
+
+class MultiOutputClassifier(BaseEstimator, ClassifierMixin, MetaEstimatorMixin):
+    """
+    This strategy consits of fitting one classifier per target. This is a
+    simple strategy for extending classifiers that do not natively support
+    multi-target classification
+
+    Parameters
+    ----------
+    estimator : estimator object
+        An estimator object implementing `fit`,`score` and `predict_proba`.
+    n_jobs : int, optional, default: 1
+        The number of jobs to use for the computation. If -1 all CPUs are used.
+        If 1 is given, no parallel computing code is used at all, which is
+        useful for debugging. For n_jobs below -1, (n_cpus + 1 + n_jobs) are
+        used. Thus for n_jobs = -2, all CPUs but one are used.
+        The number of jobs to use for the computation.
+        It does each target variable in y in parallel.
+    Attributes
+    ----------
+    estimator : Sklearn estimator
+        The base estimator used to construct the model.
+    n_outputs_ : int
+        Number of outputs in the multi-output classifier
+    """
+
+    def __init__(self, estimator, n_jobs=1):
+
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """ Fit the model to data.
+        Fits a seperate model for each output variable.
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+        y : (sparse) array-like, shape = [n_samples, n_outputs]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+        Returns
+        -------
+        self
+        """
+
+        _check_estimator(self.estimator)
+        # check to see that the X and y have the same number of rows.
+        check_consistent_length(X, y)
+        if y.ndim == 1:
+            raise ValueError("Please use the base estimator directly"
+                             "for single output classification")
+
+        self.n_outputs_ = y.shape[1]
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(delayed(_fit_classification)(
+            self.estimator, X, y[:, i]) for i in range(self.n_outputs_))
+        return self
+
+    def predict(self, X):
+        """Predict multi-output variable using a model
+         trained for each target variable.
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+        Returns
+        -------
+        y : (sparse) array-like, shape = [n_samples, n_outputs]
+            Multi-output targets predicted across multiple predictors.
+            Note:  Separate models are generated for each predictor.
+        """
+        check_is_fitted(self, 'estimators_')
```

Comment:
I feel we can use a common superclass for both `MultiOutputRegressor` and `MultiOutputClassifier` and reuse fit and predict easily